File: ai-resume-job-matcher/backend/app/schemas/job.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List


class JobBase(BaseModel):
    title: str
    company: str
    description: str
    required_skills: Optional[List[str]] = None
    location: Optional[str] = None
    salary: Optional[str] = None
    url: Optional[str] = None


class JobCreate(JobBase):
    pass


class Job(JobBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/job.ts
export interface JobCreateRequest {
  title: string;
  company: string;
  description: string;
  required_skills: string[];
  location: string;
  salary: string;
}

export interface JobResponse {
  id: number;
  title: string;
  company: string;
  description: string;
  required_skills: string[];
  location: string;
  salary: string;
}
